packages/page-council/src/Motions/Voting.tsx

```diff
@@ -42,6 +49,16 @@ function Voting ({ hash, idNumber, isDisabled, members, prime, proposal }: Props
 
   const isPrime = prime?.toString() === accountId;
 
+  const voteExtrinsic = api.tx.council.vote(hash, idNumber, voteValue);
+  const closeExtrinsic = api.tx.council.close.meta.args.length === 4
+    ? api.tx.council.close(hash, idNumber, proposalWeight, proposalLength)
+    : api.tx.council.close(hash, idNumber, 0, 0);
```

Comment:
This can't be right, see https://github.com/polkadot-js/apps/blob/master/packages/page-council/src/Motions/Close.tsx#L73-L77

So, 

- if args === 4 and passing, it needs the weights (since it will apply)
- if args === 4 and failing, the weights are 0 (it doesn't apply and we save fees)
- if args !== 4, don't pass the last 2 (here you pass them as 0, 0, so it breaks on older versions)

Reply:
Ah thanks. I actually had something more correct but super hacky (wouldn't pass linting) concerning the last point
>     if args !== 4, don't pass the last 2 (here you pass them as 0, 0, so it breaks on older versions)

I was doing something like `api.tx.council.(close as any)(hash, idNumber);`.
What would you recommend to support older versions?

Reply:
Would just do the call and override with `// @ts-ignore` for the older one. Example here - https://github.com/polkadot-js/apps/pull/3358/files#diff-e9dca1c535128a81cb2067002a8d88c5R62-R66

It is messy, but don't have a better (non-any) solution. It is valid in _some_ older cases.